### DISCUSSION TITLE
Anyone tried to connect a database from PlanetScale to a WordPress site ?

### DISCUSSION THREAD

**Discussion by sebbeauchamp** [planetscale/discussion#214] [Documentation] [answered]
Hi,
I have a hard time to connect my WordPress site to PlanetScale is there some specific things to do in the config file to remove the connection error. I have triple checked all the values. 

If someone can help, it will be really awesome !

Thx

```
/** The name of the database for WordPress */
define( 'DB_NAME', 'my-name-db' );

/** Database username */
define( 'DB_USER', 'Me-Admin' );

/** Database password */
define( 'DB_PASSWORD', 'xxxxxx••••*****!!!!!' );

/** Database hostname */
define( 'DB_HOST', 'xxxxxxxxxxxxxx.eu-west-2.psdb.cloud' );

/** Database charset to use in creating database tables. */
define( 'DB_CHARSET', 'utf8' );

/** The database collate type. Don't change this if in doubt. */
define( 'DB_COLLATE', '' );
```

**Comment by fooforge** [comment, marked as answer]
Hi @sebbeauchamp,

After adding the following configuration directive, you should be all set and able to connect:

```
define( 'MYSQL_CLIENT_FLAGS', MYSQLI_CLIENT_SSL );
```

**Reply by sebbeauchamp** [reply]
Thx @fooforge

Everything is fine now with this line

**Reply by shivapoudel** [reply]
@sebbeauchamp I tried to change the config with Local WP but unfortunately, the site is not loading. How did you configure it?

**Reply by sebbeauchamp** [reply]
Hi, have you had the line in the file wp-config ? I have configured with MAMP, maybe Local WP is blocking in someway ?

**Comment by sebbeauchamp** [comment]
Sorry @fooforge ,

I maybe though everything was find a little too quick after the installation, there was an issue with the table creation and I got this message "direct DDL is disabled" have you ever got this message ?

Thx again

**Reply by fooforge** [reply]
No worries. :-)

Have you promoted the branch to production? Those are [protected from direct schema changes](https://docs.planetscale.com/concepts/branching#development-and-production-branches) so you must create a deploy request to alter the schema. With Wordpress frequently changing the database schema (e.g. on updates or plugin installs), I would stick to development branches for the time being.

If you want your production branch to be downgraded, shoot me a quick message via https://support.planetscale.com/ from your registered email address, and I'll take care of that for you shortly.

**Reply by sebbeauchamp** [reply]
Yes it's on main branch promoted to production. I'll try few things and get back to you 

Thx

**Reply by RobinGiel** [reply]
Hi @sebbeauchamp, did you have any issues after trying? So far so good?

**Reply by scottbuscemi** [reply]
+1 to curiosity about WP success :)

**Reply by sebbeauchamp** [reply]
Hi @RobinGiel  and @scottbuscemi 

I have deployed a test with heroku and planetscale, it's only a concept for now :) I would recommend at least to try this type of environment.

**Reply by splitpierre** [reply]
I have a running project with WordPress with a fairly big database ~500mb, 98 tables, and today decided to try export and import my db into planetscale, also tried the beta import with replication but no luck.

I had all sorts of problems, most frequent one I've noticed was the NO_ZERO_DATE not allowing many of my tables to import, spent two hours fiddling around export settings but I'm nowhere near a trustable import, I'll have to go table by table and investigate each issue manually.

Planetscale instance is too good for WordPress from what I've been looking, with many kinds of plugins from different devs and skills, turns out to a **very** messy database, with an ever changing schema. So I don't think production branch fits WordPress and, not sure if it's a good fit at all considering such a messy schema.

I'll give it a second try, cause I love planetscale, and will report details on this migration process if anyone is interested, let me know.

**Reply by shivapoudel** [reply]
Since WP creates table row with default value for datetime and 0000:00:00 00:00 is place how to overcome with this NO_ZERO_DATE in planetscale.

**Reply by shivapoudel** [reply]
Excellent I found a solution to NO_ZERO_DATE for PlanetScale to run this query from client like MySQL Workbench and you will be successful:
```sql
SET sql_mode = 'ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION';
show variables like 'sql_mode'; 
```

**Comment by pau1phi11ips** [comment]
> I'll give it a second try, cause I love planetscale, and will report details on this migration process if anyone is interested, let me know.

How did you get on with this? I tried it on a WordPress install running WooCommerce, albeit on the free tier, and the performance was abysmal.10 seconds to load some pages.

**Comment by harithwick** [comment]
I keep getting this error when trying to import the wordpress db to planetscale: external database settings are not compatible with PlanetScale: "binlog_format" must be "ROW", but found: "MIXED"

has anyone come across this?

![Screenshot 2023-12-12 at 2 36 19 pm](https://github.com/planetscale/discussion/assets/25376023/7e49f076-a27d-4511-b640-ef940284d068)

**Reply by samlambert** [reply]
`mysql> SET GLOBAL binlog_format = 'ROW';` will set your originating MySQL server to ROW format.

**Reply by jgreet** [reply]
@harithwick if you haven't check it out already, the required binlog settings are laid on in our importer documentation. There are also Migration Guides for the major cloud providers that might help.

https://planetscale.com/docs/imports/database-imports#server-configuration-issues

**Comment by harithwick** [comment]
Thanks for the response. I managed to migrate the DB but i can't connect from WordPress. I keep getting this error. I've double checked my connection settings (DB_NAME, DB_USER, DB_PASSWORD, DB_HOST) in wp config and ive added the line define( 'MYSQL_CLIENT_FLAGS', MYSQLI_CLIENT_SSL ); 

![Screenshot 2023-12-13 at 9 50 57 am](https://github.com/planetscale/discussion/assets/25376023/0130c137-b374-426b-9af8-5a85befc0024)

**Reply by pau1phi11ips** [reply]
Make sure you're allowing outbound connections on port 3306

**Reply by carlHandy** [reply]
Add this to your wp-config and you should be good. However, I noticed significant slow done in my site after migrating. One thing thats odd is that memory usage on Planet Scale is bottled out at 100%.

```
define('MYSQL_CLIENT_FLAGS', MYSQLI_CLIENT_SSL);
define('MYSQL_SSL_CA', '/etc/ssl/certs/ca-certificates.crt');
```

**Reply by pau1phi11ips** [reply]
I just tried it out on the free tier and it was pretty unusable for WordPress with WooCommerce. 
1 second page loads on admin went to more than 6 seconds.

**Reply by samlambert** [reply]
Where is your Wordpress app hosted?

**Reply by carlHandy** [reply]
I'm having the exact same experience.

@samlambert mine is hosted on Digital Ocean SF03 zone. I figured since my DB is now in Ohio there'd be some latency but not this much.

**Reply by carlHandy** [reply]
@pau1phi11ips what helped my situation was strong caching, both on the DB layer and files. I used W3 Total Cache and redis on my webserver to cache queries and since I route through Cloudflare I increased the browser TTL to 1 month.

This won't be ideal if new products are added to your site frequently but it'll help a bit with load times. Tinker with caching a bit.
